nightlies_watcher/watch.py

```diff
@@ -32,61 +37,81 @@ def main(name=None):
-        last_published_task_ids_per_achitecture = json.load(f)
+def run_if_new_builds_are_present(config):
+    repository = config['repository_to_watch']
 
-    logger.debug('Latest published tasks were: %s', last_published_task_ids_per_achitecture)
+    logger.debug('Check if new builds are available')
+    latest_task_definitions_per_achitecture = get_latest_task_definitions_per_achitecture(
+        repository, config['architectures_to_watch']
+    )
+    logger.debug('Found these tasks: %s', latest_task_definitions_per_achitecture)
+
+    try:
+        revision = get_matching_revision(latest_task_definitions_per_achitecture)
+    except:
+        logger.warn('Some of the tasks are not defined against the same revision', latest_task_definitions_per_achitecture)
+        return
 
-    if have_all_tasks_never_been_published(latest_task_ids_per_achitecture, last_published_task_ids_per_achitecture):
-        logger.info('New builds found, starting publishing: %s', latest_task_ids_per_achitecture)
-        publish(latest_task_ids_per_achitecture)
+    if treeherder.does_job_already_exist(
+        repository, revision, job_name=config['task']['name'], tier=config['task']['treeherder']['tier']
+    ):
+        logger.info('Nothing to publish', latest_task_definitions_per_achitecture)
     else:
-        logger.info('Nothing to publish', latest_task_ids_per_achitecture)
+        logger.info('New builds found, starting publishing: %s', latest_task_definitions_per_achitecture)
+        publish(config, revision, latest_task_definitions_per_achitecture)
 
 
-def get_latest_task_ids_per_achitecture(android_architectures):
-    return {
+def get_latest_task_definitions_per_achitecture(repository, android_architectures):
+    task_ids_per_achitecture = {
         pusk_apk_architecture_name: {
-            'task_id': get_latest_task_id(namespace_architecture_name)
+            'task_id': get_latest_task_id(repository, namespace_architecture_name)
         }
         for pusk_apk_architecture_name, namespace_architecture_name
         in android_architectures.items()
     }
 
+    return {
+        architecture_name: {
+            'task_id': data['task_id'],
+            'revision': get_revision(data['task_id'])
+        }
+        for architecture_name, data
+        in task_ids_per_achitecture.items()
+    }
 
-def get_latest_task_id(android_architecture):
-    namespace = get_full_name_space(android_architecture)
-    task = index.findTask(namespace)
-    return task['taskId']
 
+def get_matching_revision(tasks_per_architecture):
+    revisions = [task['revision'] for _, task in tasks_per_architecture.items()]
+    if len(revisions) == 0:
+        raise Exception('No tasks given')
 
-def get_full_name_space(android_architecture):
-    return 'gecko.v2.mozilla-aurora.nightly.latest.mobile.{}'.format(android_architecture)
+    first_revision_to_compare_to = revisions[0]
+    if (all(revision == first_revision_to_compare_to for revision in revisions)):
```

Comment:
I don't believe so. We want to make sure that every revision is equal to the others in the list. `first_revision_to_compare_to in revisions` is always true since the first revision is the first item of the list.